### PR TITLE
Encoder decoder config docs

### PR DESCRIPTION
**Pull request by afcruzs**
As discussed on #5826, this PR adds more details on how to load encoder/decoder config objects from pretrained folders and how to instantiate encoder_decoder pretrained models given their corresponding configuration objects (useful for loading pre-trained models and modifying some config members for fine-tuning).

### CODE REVIEW

File: src/transformers/configuration_encoder_decoder.py [hunk 1]
```diff
@@ -56,6 +56,13 @@ class EncoderDecoderConfig(PretrainedConfig):
             >>> # Accessing the model configuration
             >>> config_encoder = model.config.encoder
             >>> config_decoder  = model.config.decoder
+
+            >>> Saving the model, including its configuration
+            >>> model.save_pretrained('my-model')
+
+            >>> # loading model and config from pretrained folder
+            >>> encoder_decoder_config = EncoderDecoderConfig.from_pretrained('my-model')
+            >>> model = EncoderDecoderModel.from_pretrained('my-model', config=encoder_decoder_config)
```
Comment (patrickvonplaten) [inline src/transformers/configuration_encoder_decoder.py:65]:
```suggestion
            >>> model = EncoderDecoderModel.from_pretrained('my-model')
```

Comment (patrickvonplaten) [reply src/transformers/configuration_encoder_decoder.py:65]: no need to add the config here

File: src/transformers/modeling_encoder_decoder.py [hunk 2]
```diff
@@ -128,6 +128,15 @@ def from_encoder_decoder_pretrained(
 
             >>> from transformers import EncoderDecoderModel
             >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained('bert-base-uncased', 'bert-base-uncased') # initialize Bert2Bert
+
+        Example loading pretrained configs separately::
+            >>> from transformers import BertConfig, EncoderDecoderModel
+
+            >>> pretrained_model_name_or_path = 'bert-base-multilingual-cased'
+            >>> encoder_config = BertConfig.from_pretrained(pretrained_model_name_or_path)
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:136]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 3]
```diff
@@ -128,6 +128,15 @@ def from_encoder_decoder_pretrained(
...
+            >>> decoder_config = BertConfig.from_pretrained(pretrained_model_name_or_path, is_decoder=True)
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:137]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 4]
```diff
@@ -128,6 +128,15 @@ def from_encoder_decoder_pretrained(
...
+            >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained(pretrained_model_name_or_path, pretrained_model_name_or_path, encoder_config=encoder_config, decoder_config=decoder_config)
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:138]:
```suggestion
            >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained(pretrained_model_name_or_path, pretrained_model_name_or_path)
```

**Review by patrickvonplaten — COMMENTED** [review]
Example of the model can be simplified => no need to set `decoder = True`, since this is done automatically.

File: src/transformers/modeling_encoder_decoder.py [hunk 5]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
 
             >>> from transformers import EncoderDecoderModel
             >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained('bert-base-uncased', 'bert-base-uncased') # initialize Bert2Bert
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:130]:
```suggestion
            >>> # initialize a bert2bert from two pretrained BERT models. Note that the cross-attention layers will be randomly initialized
            >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained('bert-base-uncased', 'bert-base-uncased') 
            >>> # saving model after fine-tuning
            >>> model.save_pretrained("./bert2bert")
            >>> # load fine-tuned model
            >>> model = EncoderDecoderModel.from_pretrained("./bert2bert")
```

File: src/transformers/modeling_encoder_decoder.py [hunk 6]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+
+        Example loading pretrained configs separately::
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:132]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 7]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+            >>> from transformers import BertConfig, EncoderDecoderModel
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:133]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 8]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:134]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 9]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+            >>> pretrained_model_name_or_path = 'bert-base-multilingual-cased'
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:135]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 10]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+            >>> model = EncoderDecoderModel.from_encoder_decoder_pretrained(pretrained_model_name_or_path, pretrained_model_name_or_path)
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:136]:
```suggestion
```

File: src/transformers/modeling_encoder_decoder.py [hunk 11]
```diff
@@ -128,6 +128,13 @@ def from_encoder_decoder_pretrained(
...
+
```
Comment (patrickvonplaten) [inline src/transformers/modeling_encoder_decoder.py:137]:
```suggestion
```

File: src/transformers/configuration_encoder_decoder.py [hunk 12]
```diff
@@ -56,6 +56,13 @@ class EncoderDecoderConfig(PretrainedConfig):
             >>> # Accessing the model configuration
             >>> config_encoder = model.config.encoder
             >>> config_decoder  = model.config.decoder
```
Comment (patrickvonplaten) [inline src/transformers/configuration_encoder_decoder.py:58]:
```suggestion
            >>> config_decoder  = model.config.decoder
            >>> # set decoder config to causal lm
            >>> config_decoder.is_decoder = True
```

Comment (patrickvonplaten) [reply src/transformers/configuration_encoder_decoder.py:58]: show how user can change  decoder config manually

**Comment by patrickvonplaten** [comment]
@afcruzs - thanks a lot for your PR. I changed the examples a bit trying to make sure:
1) Examples in `config` only concern the EncoderDecoderConfig
2) Examples in `model` only concern the EncoderDecoderModel

sorry for meddling in your PR.

File: src/transformers/configuration_encoder_decoder.py [hunk 13]
```diff
@@ -56,6 +56,15 @@ class EncoderDecoderConfig(PretrainedConfig):
...
+            >>> # set decoder config to causal lm
+            >>> config_decoder.is_decoder = True
+
+            >>> Saving the model, including its configuration
```
Comment (patrickvonplaten) [inline src/transformers/configuration_encoder_decoder.py:62]:
```suggestion
            >>> # Saving the model, including its configuration
```

**Comment by afcruzs** [comment]
Thanks for improving the examples! I've fixed the whitespace issue

**Pull request merged**